internal/resource/helper.go

```diff
@@ -56,6 +57,12 @@ func parseURL(s *v1.Service) error {
 	return nil
 }
 
+func isReference(v string) bool {
+	// https://github.com/Kong/kong/blob/5d721ac9ae1df36049013599d0253bd39cb6f758/kong/pdk/vault.lua#L320-L328
+	r := regexp.MustCompile(typedefs.ReferencePattern)
```

Comment:
Can we move this outside the function as an un-exported variable, to not compile the same regex every time? 

The other advantage is if `typedefs.ReferencePattern()` is an invalid regex, it'll panic almost immediately during application init instead of at function runtime (the `Must*()` functions generally are reserved for this exact use-case, as otherwise, we wouldn't want any possibility of panics).

Reply:
good catch, done.